### PR TITLE
Expose .thenStop on ReplyEffects

### PR DESCRIPTION
**Pull request by leviramsey**
The various persistence DSL's `ReplyEffect`s do not have a `thenStop` method, so in a reply-enforcing behavior, to stop the actor one must use something like:

```java
return Effect().persist(someEvent).thenStop().thenReply(replyTo, updatedState -> replyMsg);
```

which gives the impression that the actor stops and then replies "from beyond the grave".  Presumably the interpreter of the effect defers the stop to be the last side-effect, in which case the `then`s suggest an ordering that isn't true (if not, then expressing "send a final reply and stop" would need something infelicitous like `Effect().persist(someEvent).thenRun(updatedState -> replyTo.tell(replyMsg)).thenStop().thenNoReply()`).

Since the concrete implementations of the `Effect`s already have `thenStop` methods, this change exposes them through the `ReplyEffect` interfaces as well as `EffectBuilder`.

### CODE REVIEW

File: akka-persistence-typed/src/main/mima-filters/2.9.6.backwards.excludes/32516-ReplyEffect-thenStop.excludes [hunk 1]
```diff
@@ -0,0 +1 @@
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.persistence.typed.scaladsl.ReplyEffect.thenStop")
```
Comment (patriknw) [inline akka-persistence-typed/src/main/mima-filters/2.9.6.backwards.excludes/32516-ReplyEffect-thenStop.excludes:1]: mima file should have the latest release in the name, i.e. 2.9.5

**Review by patriknw — COMMENTED** [review]
looking good

File: akka-persistence-typed/src/main/mima-filters/2.9.6.backwards.excludes/32516-ReplyEffect-thenStop.excludes (hunk 1 above)
Comment (leviramsey) [reply akka-persistence-typed/src/main/mima-filters/2.9.6.backwards.excludes/32516-ReplyEffect-thenStop.excludes]: Weird... one of the commits had it the mima in `2.9.5.backwards.excludes` and mima rejected (https://github.com/akka/akka/pull/32516/commits/07aa5a9212337cb7993a22b4a62fb7dd0dbc1beb)

Comment (leviramsey) [reply akka-persistence-typed/src/main/mima-filters/2.9.6.backwards.excludes/32516-ReplyEffect-thenStop.excludes]: Ah that's it... the first time, it just complained about event-sourced

**Review by johanandren — APPROVED** [review]
LGTM

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**